ringersong/src/main/java/com/RingerSong/free/data/YouTubeMusicRepository.kt

```
@@ -216,4 +217,96 @@ class YouTubeMusicRepository(private val context: Context) {
+
+            val localFile = java.io.File(musicDir, "$videoId.mp3")
+
+            audioResponse.body?.byteStream()?.use { input ->
+                java.io.FileOutputStream(localFile).use { output ->
+                    input.copyTo(output)
+                }
+            }
+
+            Log.d(TAG, "Downloaded YouTube track: ${localFile.absolutePath}")
+            return@withContext DownloadResult.Success(localFile.absolutePath)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)